ruma-push-gateway-api/src/notify/send_event_notification/v1.rs

```diff
@@ -138,6 +139,7 @@ impl<'a> Notification<'a> {
     /// * the content of the notification
     /// * the number of unacknowledged communications for the recipient
     /// * the devices to send the notification to
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
```

Comment:
Only the `Required` fields need to be included in the constructor or just use `Self::default`. The idea for non_exhaustive is to use a macro [assign](https://crates.io/crates/assign) to allow for more natural-looking declarations.

Reply:
Gotcha, sounds good switched to only required parameters.